spec/unit/defines/concat_fragment_spec.rb

```diff
@@ -46,13 +46,21 @@
       should contain_file("#{fragdir}/fragments/#{p[:order]}_#{safe_name}").with({
         :ensure  => safe_ensure,
         :owner   => id,
-        :group   => gid,
         :mode    => '0640',
         :source  => p[:source],
         :content => p[:content],
         :alias   => "concat_fragment_#{title}",
         :backup  => false,
       })
+      # The defined() function doesn't seem to work properly with puppet 3.4 and rspec.
+      # defined() works on its own, rspec works on its own, but together they
+      # determine that $gid is not defined and cause errors here. Work around
+      # it by ignoring this check for older puppet version.
+      if Puppet::Util::Package.versioncmp(Puppet.version, '3.4.0') > 0
```

Comment:
This should account for puppet 3.4.1 - 3.4.3 (the latest 3.4 release). Perhaps you could do `Puppet::Util::Package.versioncmp(Puppet.version, '3.5.0') < 0`